src/templates/next/components/shared/CollectionSearch/CollectionSearch.tsx

```diff
@@ -0,0 +1,29 @@
+import { BiSearch } from "react-icons/bi"
+import type { CollectionSearchProps } from "../../../types/CollectionSearch"
+
+const CollectionSearch = ({
+  placeholder,
+  search,
+  setSearch,
+}: Omit<CollectionSearchProps, "type">) => {
+  return (
```

Comment:
not entirely sure if it's possible but we might benefit from trying to debounce here? 

something we guard against is users typing in rapidly and we trigger researches on each intermediate query (eg: a-> ab -> abb -> abba)

Reply:
Yeah I think it's a good point, but not going to optimise things for now until we integrate everything together